feat: added loader for Provider install progress

rogueloop opened the pull request:
# Loader to show provider install progress

## Description

Added a loader for provider installation, for some feedback to the user. Added a message parameter to the WithSpinner() function to pass the loader message.

## Related Issue(s)

closes #734
claim #734

## Screenshots

New Install message 
[Kazam_screencast_00002.webm](https://github.com/user-attachments/assets/71cb4ac5-ca51-4392-84ac-ef7c4fd8158b)

lbrecic reviewed (changes requested):
Hi @rogueloop, thanks for your contribution!
Code looks fine and everything works, just a few following changes should be made before we can proceed with this:

Please add appropriate Pull Request Title in the description part and upload the image directly to the github instead of the google drive link.

Also, in the Related Issue(s) part, please instead of `This PR addresses issue #734` write `Closes #734`.
`/closes #734` can be removed as well.

rogueloop commented:
> Hi @rogueloop, thanks for your contribution! Code looks fine and everything works, just a few following changes should be made before we can proceed with this:
> 
> Please add appropriate Pull Request Title in the description part and upload the image directly to the github instead of the google drive link.
> 
> Also, in the Related Issue(s) part, please instead of `This PR addresses issue #734` write `Closes #734`. `/closes #734` can be removed as well.
> 
> Formatting check is failing so please tend to that as well. 😄

Addressed the issues :+1:

lbrecic reviewed (approved):
Nice! Thanks for the contribution! 😄

idagelic commented on pkg/cmd/provider/install.go:
```diff
@@ -77,16 +78,25 @@ var providerInstallCmd = &cobra.Command{
-		if err != nil {
-			log.Fatal(apiclient_util.HandleErrorResponse(res, err))
-		}
+		err = view_utils.With(func() error {
```
I just noticed some confusing naming in the existing spinner package and changed this function to "WithSpinner" to make it more clear. Please do a rebase from main to fix this.

One change I would like to propose is to change the text from "Loading" to "Installing". I believe an easy way to do this is to add a property to the view model and have the label passed into WithSpinner

rogueloop replied:
Added the changes pls check @idagelic.

rogueloop commented:
Added the changes pls check @idagelic.

idagelic reviewed (changes requested):
This looks good now, thanks.

There is a check failing - I believe it is because of code formatting - please try to resolve it by running `go fmt ./...`

rogueloop commented:
> This looks good now, thanks.
> 
> There is a check failing - I believe it is because of code formatting - please try to resolve it by running `go fmt ./...`

Done :heavy_check_mark:
